File: 240229/개인정보/personal-info.py
class Student:
    def __init__(self, n, h, w):
        self.n, self.h, self.w = n, h, w

students = []
for _ in range(5):
    n, h, w = tuple(input().split())
    students.append(Student(n, int(h), float(w)))

students.sort(key = lambda x: x.n)

print("name")
for student in students:
    print(student.n, student.h, student.w)

print()

students.sort(key = lambda x: -x.h)

print("height")
for student in students:
    print(student.n, student.h, student.w)

File: 240229/개인정보/personal-info.js
class Person {
    constructor(n, h, w) {
        this.n = n;
        this.h = h;
        this.w = w;
    }
}

const fs = require("fs");
let inputs = fs.readFileSync(0).toString().trim().split("\n");

const persons = []
for(const p of inputs) {
    const person = p.trim().split(" ")
    persons.push(new Person(person[0], Number(person[1]), Number(person[2])))
}

persons.sort((a,b) => b.n > a.n ? -1 : 1)
console.log('name')
for(const p of persons) {
    console.log(`${p.n} ${p.h} ${p.w}`)
}

console.log()
persons.sort((a,b) => b.h - a.h)
console.log('height')
for(const p of persons) {
    console.log(`${p.n} ${p.h} ${p.w}`)
}